lib/SILOptimizer/Utils/ConstantFolding.cpp

```diff
@@ -995,6 +995,8 @@ struct IEEESemantics {
 
 IEEESemantics getFPSemantics(BuiltinFloatType *fpType) {
   switch (fpType->getFPKind()) {
+  case BuiltinFloatType::IEEE16:
```

Comment:
Could you remove the `default` below and expand things out here too? (Not that I anticipate us supporting more exotic FP formats than Power).

Reply:
Why? We could certainly do it, but it's not clear to me how that benefits us.

Reply:
I just hit a bizarre crash in the middle of building a client project that tried to enter constant prop for Float16, found this code, and realized that the default provides zero signal for future authors that need to go update the supported formats - like you!

Serendipity leads to nits! 😅

Reply:
Can you sketch out what the change you'd like to see actually is? It's not totally obvious to me how you want to improve this.

Reply:
```
IEEESemantics getFPSemantics(BuiltinFloatType *fpType) {
  switch (fpType->getFPKind()) {
  case swift::BuiltinFloatType::IEEE16:
    return IEEESemantics(16, 5, 10, false);
  case BuiltinFloatType::IEEE32:
    return IEEESemantics(32, 8, 23, false);
  case BuiltinFloatType::IEEE64:
    return IEEESemantics(64, 11, 52, false);
  case BuiltinFloatType::IEEE80:
    return IEEESemantics(80, 15, 63, true);
  case swift::BuiltinFloatType::IEEE128:
    return IEEESemantics(128, 15, 112, false);
  case swift::BuiltinFloatType::PPC128:
    llvm_unreachable("128-bit PowerPC floats are not yet supported.");
  }
  llvm_unreachable("Unexpected FP semantics");
}
```